[MM-40610] add 5px right border to allow scrollbars

kelderek opened the pull request:
#### Summary
Add a 5px right hand border to allow use of scrollbars in the desktop app.  Per MM-40610 the scrollbar component used in the desktop app is very narrow and almost completely overlaps the Windows area used for grabbing the side of the window to resize it.  This PR shrinks the post-list and admin console div by 5 pixels to avoid that.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-40610

#### Related Pull Requests
None

#### Screenshots
Desktop app/Chrome Post list before:
![image](https://user-images.githubusercontent.com/6126101/198672529-4d0c8a08-096a-4765-af29-d0d7c2c84050.png)

Desktop app/Chrome Post list after:
![image](https://user-images.githubusercontent.com/6126101/198672090-97fdb831-2ea7-4f40-aa17-06100b8c29b9.png)

Firefox Post list before:
![image](https://user-images.githubusercontent.com/6126101/198671283-ec3e4b9e-9816-43f3-b42c-184efe3f18e1.png)

Firefox Post list after:
![image](https://user-images.githubusercontent.com/6126101/198670461-997c7b2c-0105-484d-a012-5014c68eee16.png)

Desktop app/Chrome System console before:
![image](https://user-images.githubusercontent.com/6126101/198673159-a52eeba1-bf07-482b-91de-e3038fcfa41d.png)

Desktop app/Chrome System console after:
![image](https://user-images.githubusercontent.com/6126101/198672294-37df9d51-45b6-4b72-b072-f1cd3c5975c9.png)

Firefox System console before:
![image](https://user-images.githubusercontent.com/6126101/198671594-22f03f6e-3e8e-4fce-b867-76ac18ac1e9a.png)

Firefox System console after:
![image](https://user-images.githubusercontent.com/6126101/198670166-d153e46e-c5e0-40af-8c14-371887510f8c.png)

#### Release Note
```release-note
-Moved scrollbars slightly away from the edge of the window to make it easier to grab them on the desktop app 
-Restyled many of the native scrollbars to more closely match the custom scrollbars
```

mattermod commented:
Hello @kelderek,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

Per the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/), we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.

**Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?**
Once you have signed the CLA, please comment with `/check-cla` and confirm that the CLA check is green.

This is a standard procedure for many open source projects.

Please let us know if you have any questions.

We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

esethna reviewed (changes requested):
@kelderek looks like the same needs to be applied to the right-hand side and thread viewer:

![image](https://user-images.githubusercontent.com/14333569/194401327-a9f1b666-4df6-4fba-81d3-ecbb81d452db.png)
![image](https://user-images.githubusercontent.com/14333569/194401471-1f088d8e-45b9-45ef-a304-0c6915e41d95.png)

kelderek commented:
Good call @esethna, I forgot about that.  Will take a look at that presently

kelderek commented:
Not sure why the test server creation failed messages are coming up or how to view the logs the message references.  Maybe those are delayed from the times when I made the linter angry?

Also looking at Boards and Playbooks, similar adjustments would have to be made there, but it looks like those are separate Github repos so would have to be separate PRs, if I understand correctly.

esethna commented:
> Also looking at Boards and Playbooks, similar adjustments would have to be made there, but it looks like those are separate Github repos so would have to be separate PRs, if I understand correctly.

Good question. @devinbinnie thoughts? How can we ensure we've gotten all the locations that could be impacted by this ux-defect?

esethna reviewed (approved):
Looks great! @jgilliam17 please try to check all locations where a right scroll bar may appear on the right edge of the app

Note: I don't see a scroll bar anymore for the saved posts list, but I cannot confirm if that is due to this pr or something odd on the test server cc// @jgilliam17

esethna commented:
@kelderek appreciate you jumping in to fix this. And as a bonus, it's hacktoberfest!

kelderek commented:
@esethna Are pinned posts the same as saved posts?  I don't see scrollbars on the pinned posts or search boxes in my vanilla 7.3.0 instance.  I posted back on https://github.com/mattermost/desktop/issues/1513 about that - they are actually there but being hidden by the margin-right property.

If you think it makes sense, I could try to add changes to those on this PR, or start another.

esethna commented:
@kelderek **pinned posts** (posts pinned in a specific channel), **saved posts** (posts saved by you across all your channels), **recent mentions** (mentions of your username across all your channels), and **search** all use the right-hand side pane for displaying results

kelderek commented:
Ah ok I see now.  I will check on all four of those

kelderek commented:
turns out the scrollbars are actually there (the same auto-hiding ones as on the channel list left hand sidebar) but its color was the same as the background so it was invisible.  Working on changing the color and position of the scrollbar now.  Almost there but I have it applied to the wrong spot.  I am done for today, but will pick this back up this weekend.  I think I am almost there!

kelderek commented:
Found and fixed a graphical issue with using the right property in the thread view.  When using a wide enough layout for threads to show up as a third column, because the right property shifted everything the text edit box in the final div was overlapping and hiding the left hand vertical border.  Switched to using margin-right instead which doesn't have that problem

esethna commented:
+1 Thanks for the detail above, 100% agree with your future considerations, really appreciate the work you've done here. Will leave it in the hands of dev and QA for final reviews. Thanks again! @kelderek

kelderek commented:
All the checks passed and I have run through things in the desktop app and Firefox one last time and all looks good.  Will be hands off now unless requested otherwise.  Thanks @esethna

AshishDhama commented on components/search_results/search_results.tsx:
```diff
@@ -39,6 +39,12 @@ import './search_results.scss';
 
 const GET_MORE_BUFFER = 30;
 
+const scrollbarStyles = {
```
@kelderek Instead of defining styles, would be better if define a class and use that

kelderek replied:
Ok, I will give it a shot.  I originally did it this way because that is how it was done in post_list_virtualized.tsx

kelderek replied:
done

AshishDhama commented on components/search_results/search_results.tsx:
```diff
@@ -49,13 +55,15 @@ const renderView = (props: Record<string, unknown>): JSX.Element => (
 const renderThumbVertical = (props: Record<string, unknown>): JSX.Element => (
     <div
         {...props}
+        style={scrollbarStyles}
```
```suggestion
     
```

kelderek replied:
done

AshishDhama commented on components/search_results/search_results.tsx:
```diff
@@ -49,13 +55,15 @@ const renderView = (props: Record<string, unknown>): JSX.Element => (
     <div
         {...props}
+        style={scrollbarStyles}
         className='scrollbar--vertical'
```
```suggestion
        className='scrollbar--vertical scrollbar--position'
```
we can define this class where have defined `scrollbar--vertical`

AshishDhama commented:
> I hit re-request review for AshishDhama after making their suggested changes, but it looks like it removed the review request for jgilliam17 Apologies if I wasn't supposed to his re-request review; I am new to the github PR review process. I assume it will re-add jgilliam17 or do you need to do that, @esethna ?

@kelderek Hey no worries it’s a bug or maybe a feature, not your fault. Thanks for working on the feedback.

kelderek commented on components/post_view/post_list_virtualized/post_list_virtualized.tsx:
```diff
@@ -44,6 +44,7 @@ const postListStyle = {
 const virtListStyles = {
     position: 'absolute',
     bottom: '0',
+    right: '5px',
```
margin-right doesn't work in this case, I think it has to do with the positioning of the element.  For whatever reason the post list uses the native toolbars and not the custom toolbars the sidebars use, so I had to approach it differently and move the entire div over slightly.

kelderek replied:
That isn't an ideal solution as it doesn't just shift over the scollbar, it moves the posts over as well.  Looking at it again now, I might be able to change it up and modify the post-list__content class with a margin-right instead.  That seems to just move over the scrollbar.  I will try that out and look for anomolies

kelderek replied:
@matthewbirtch It looks like removing the "right" property from here and adding "margin-right" to .post-list__table .post-list__content did the trick and is a better solution.  I tested it out and couldn't find any regressions from the move.  Thanks for the feedback!

kelderek commented:
@matthewbirtch Change based on your feedback has been implement and it is ready for review again.  Thanks!

matthewbirtch commented:
Looking at this from a UI standpoint, we now have an awkward space created here caused by the extra 5px gap added on the right of the scrollable panes. This doesn't look great, unfortunately. I wonder if we remove the scrollbar track background if that will help with this awkwardness. 

If we do remove the track background, it might create other problems with the hover states for posts not spanning the full width.

kelderek commented:
I definitely hear you @matthewbirtch Here are my two cents on the topic:

1) It definitely felt kinda of bad and kludgy at first.  After having lived with it for a bit, I no longer think it looks weird and the very thin scrollbar in the app on the far right edge now looks weird to me.  Browser versions look fine to me as well, although I have to admit with Firefox's chunkier scrollbars do look a bit odd not being on the right.  There is a CSS property scrollbar-width, as far as I can tell it is only supported on Firefox 64+ anyway, that can be set to thin making Firefox look basically like the app and other browsers.

2) As shown in some screenshots here https://github.com/mattermost/desktop/issues/1513#issuecomment-1103210555 Slack's app apparently also has a gap, although theirs looks smaller

3) We could potentially change to using the custom scrollbar for the post list rather than the native scrollbar, which would also improve consistency within the app.  Not sure how big a lift that would be.

kelderek commented:
For reference, here is what Firefox looks like with scrollbar-width: thin
![image](https://user-images.githubusercontent.com/6126101/196231252-830cb6b0-56ea-4757-bf20-5250340d0835.png)

matthewbirtch commented:
@kelderek I think it would look better without the track background color though. Is that something you could give a try?

> 3\. We could potentially change to using the custom scrollbar for the post list rather than the native scrollbar, which would also improve consistency within the app.  Not sure how big a lift that would be.

This would be great, but I'm not sure what would be involved and the level of effort we'd be talking about.

kelderek commented:
@matthewbirtch Agreed, it would be better without the track visible.  All the other systems had an invisible track as well.

I looked into it a bit earlier today.  For webkit based systems like chrome, edge, and the desktop app we could do it easily by changing the the -webkit-scrollbar-track alpha in from the 0.1 it is now to 0.  That will make it invisible.  Firefox would need an alternative solution and the only thing I have been able to do so far is change its color but not make it transparent.  If we match it to the theme color then maybe that is viable, but it certainly isn't may favorite approach.

I will take a look tomorrow to see if moving to the custom scrollbar is something I am capable of doing.  Even if I can't, I think that even though this PR isn't perfect, it represents a step forward. Especially for users of the desktop app where there the area for clicking the scrollbar is currently only 1 or 2 pixels wide.

matthewbirtch commented:
> Firefox would need an alternative solution and the only thing I have been able to do so far is change its color but not make it transparent. If we match it to the theme color then maybe that is viable, but it certainly isn't may favorite approach.

If you're using hex values for firefox scrollbar colors, you should be able to set the alpha to 0 by adding `00` on the end of your hex color. For example: `#FFFFFF00` would be transparent white

kelderek commented:
Thanks; adding the trailing zeros works!  I checked on some of the themes and there are at least two different colors for the part of the scrollbar that actually gets dragged, so now I just need to find the variable the themes use for that so the bars stay theme consistent

AshishDhama commented:
@kelderek you can use `make fix-style` to fix lint issues.
I have one more suggestion if that makes sense. Use this mixin at other places as well.

```scss
@mixin scrollbar-position-mixin{
    position: relative;
    right: 5px;
    background-color: rgba(var(--center-channel-color-rgb), 0.32);
}
.scrollbar--position {
@include scrollbar-position-mixin;
}

kelderek commented:
Thanks @AshishDhama  - I am an admin by trade and haven't done any real coding in 20+ years so I don't know the current practices.  How do I use make fix-style?  Do I just post it as a comment here and the bot will pick it up?

AshishDhama commented:
@kelderek Oh sorry I just assumed, it’s a custom command we have built as an alternative to `npm run fix`
you can find a bunch of other commands here https://github.com/mattermost/mattermost-webapp/blob/master/Makefile.

AshishDhama commented:
> Thanks @AshishDhama - I am an admin by trade and haven't done any real coding in 20+ years so I don't know the current practices. How do I use make fix-style? Do I just post it as a comment here and the bot will pick it up?

@kelderek On a side note we can help you change that, please feel free to reach out on our community server https://community.mattermost.com/ if you are interested in open source.

kelderek commented:
@AshishDhama Ok, I think that might not apply in my use case.  Every commit was supposed to be a a quick one so I have literally been doing everything via editing the source on github directly.  I keep thinking I don't actually need to setup a local dev environment because it's just one more thing haha.  That's why there are so many rapid commits which I am sure is poor coding form, sorry!

AshishDhama commented:
@kelderek we have gitpod setup which runs like a local setup, You can try that next time https://github.com/mattermost/mattermost-gitpod-config (if that works for you).
> That's why there are so many rapid commits which I am sure is poor coding form, sorry!

It's totally fine. Thanks for the contribution you are doing a great job.

kelderek commented:
@AshishDhama Thanks for all the help and encouragement!  There is a lot going on with Gitpod and it took a minute to get my head around it, but once I did it was super helpful!  I was able to use it to test a bunch of stuff way more rapidly than I had been doing, and I even used `make fix-style` to fix yet another linter error on parameter order!

@matthewbirtch I haven't yet looked at the custom scrollbars, but I have the thin scrollbars on Firefox now to better match other browsers, and transparent scrollbar tracks implemented for Firefox and Webkit browsers.  I think it is ready for you to take another look to see if you feel it is a net improvement over the current design.  Thanks!

matthewbirtch commented:
Thanks @kelderek. I just noticed that the scrollbar for the Saved Posts right side pane behaves differently than the thread view right side pane? I assumed all RHS views would be the same, but it looks like that is not the case right now. We should make sure all Right panes behave the same with the scrollbar. Is this something you could take a look at? Or maybe @AshishDhama could help identify those as well.

- Search Results RHS
- Saved Posts RHS
- Pinned Posts RHS
- Recent Mentions RHS
- Channel Info RHS
- Channel Members RHS

Also, now the threads list items in the `Threads` inbox view don't span the full width because of the scrollbar. Ideally, we find a way for the scrollbar to sit on top of everything without having to push all content to the left like Slack does, but it sounds like that requires a custom scrollbar?

<img width="679" alt="image" src="https://user-images.githubusercontent.com/2040554/196444048-537ad67b-922f-4c6a-bb05-937b6d211e87.png">

Also, I noticed the firefox scrollbar width is 6px - I like the narrower bar. Can we make Chrome's the same?

kelderek commented:
@matthewbirtch You are correct that the scrollbars are different, that is because some are native scrollbars and some are custom scrollbars.  The quickest way to tell which is which for Mattermost is the custom ones are all set to hide after 0.5 seconds when the mouse isn't over them.  Getting them all to be consistent, which I agree would be ideal, would require switching out the native ones for custom ones.  I will try to get a feel for how easy/hard that will be today.  My gut is that it would probably be non-trivial and would be better in a separate PR, but we'll see how it goes.  Luckily there are plenty of places it is used so already that I can reference.

You are also correct about the threads list that this does shrink it by 5px for the margins, and afaik the only way around would be the custom scrollbars because they sit on top of the content in their own div, whereas the native ones are a part of the content div.  This also applies to the post list, system console, and anywhere else native scrollbars are used.

I believe the Chrome toolbar size change should be very quick and easy, I will take a stab at that shortly.

kelderek commented:
@matthewbirtch I changed the scrollbar width to 6px for webkit browsers like Chrome and the mattermost app, but I have mixed feelings about it.  On the one hand, I agree that it looks nicer but overall and matches the custom scrollbars.  On the other hand I feel like usability takes a hit because the target is smaller.  Even with that, the 5px right hand border I added still gives the desktop app users a significantly bigger target than they had before, so still a net win.

I do find it surprising you said that the scrollbar on Firefox was 6px.  On my Windows 10 system, the Firefox thin scrollbar is 8px (what Chrome was before my last change). Are you on a Mac by any chance?  It appears that Firefox is using OS specific widgets.

When I looked it at on my Kubuntu linux machine at home, which has narrow scrollbar widgets, the regular size Firefox scrollbar was already thin (prob 6px, didn't check it) and the thin setting made it extremely thin, probably only one or two px.  Luckily when you roll over it, it changes to a wider scrollbar, but even then it is still thinner than Firefox on Windows.  I do have an old Mac I could try things out on, but it no longer gets MacOS updates so I don't know if it would represent what current Mac users see.

matthewbirtch commented:
> I do find it surprising you said that the scrollbar on Firefox was 6px. On my Windows 10 system, the Firefox thin scrollbar is 8px (what Chrome was before my last change). Are you on a Mac by any chance? It appears that Firefox is using OS specific widgets.

Yep, I'm on mac

kelderek commented:
@matthewbirtch I took a stab at trying to add the custom scrollbar to the main post list and didn't have any success.  It is definitely a non-trivial operation for an amateur like me.  I agree that ultimately it should be done, but I think it would make more sense as its own issue.  

Thanks to everyone's help, I have been able to expand this PR to include setting the size, shape and color of the native scrollbars to better match the custom scrollbars.  I think that moving the remaining panes away from the native scrollbars is out of scope for this PR, and beyond my time and skills at the moment.

What are your thoughts on moving forward with this PR without fully standardizing on the custom scrollbars?

kelderek commented:
@AshishDhama @matthewbirtch Just wondering where things are with this PR; is it still being evaluated?  It has been a week since I heard anything about it.

matthewbirtch commented:
Sorry for the delay @kelderek. There have been some internal discussions about this and whether it's the right approach or not. I also noticed a few places where this creates awkwardness in the layouts:

1. In the `Threads` view, the bottom border of the Thread header in the preview pane does not span all the way to the right edge now (due to the 5px margin that's been added)

<img width="1521" alt="Screen Shot 2022-10-26 at 3 56 56 PM" src="https://user-images.githubusercontent.com/2040554/198124662-74bd20c7-f033-464b-aa08-d6e6296cf7aa.png">

2. Also in the `Threads` view, in the inbox list when you hover over an item the background does not span the full width due to the new scrollbar style which feels a bit awkward as well. (see screenshot above)

3. When viewing a Thread in the RHS, if you hover over a message, the background does not span the full width due to the new scrollbar style which feels a bit awkward as well.
<img width="434" alt="Screen Shot 2022-10-26 at 3 59 24 PM" src="https://user-images.githubusercontent.com/2040554/198125079-2324a1b0-071e-4ce7-bd1e-a317e06f5abe.png">

It's very possible there are other areas affected, but these are the ones I noticed

kelderek commented:
Thanks for getting back to me!  I just didn't want to get lost in the shuffle.    To address your specific points:

1) It looks like if I move the margin-right from .ThreadPane into .ThreadViewer, that clears up the line not going all the way over when viewing a thread from the thread.  Unfortunately it messes up viewing a thread from the post list.   .post-right-comments-container may be another option but I haven't tested it fully yet
Thread list:
![image](https://user-images.githubusercontent.com/6126101/198137942-ee9c9a6e-623d-4fbc-a056-d3509230161e.png)
Post list:
![image](https://user-images.githubusercontent.com/6126101/198137700-22c780ff-c706-4ab4-b0bb-6b473ecabb72.png)

2 & 3) Fair points.  It doesn't look bad when up against a scrollbar, but you are right that overall it is not ideal

As for whether it is the right approach or not, I think it is and isn't.  For the custom scrollbars, I think it works and shouldn't cause any problems because it just floats over the content div.  The places which are using native scrollbars are where we run into problems like the ones you outlined because the scrollbar is part of the content.  The right approach for those would be to convert them over to custom scrollbars to match the other parts of the system.  That is unfortunately beyond my current time and/or ability.

Unless the full conversion to custom scrollbars is imminent, I still feel that this a useful stop-gap until someone implements custom scrollbars.  Personally, I still think the pros outweigh the cons, but if you all decide this isn't the way you want to go, just let me know.  Thanks!

Here are the pros as I see them:
-Usability gain of having a reasonable amount of pixels available to grab the scrollbar in the desktop app
-Makes the scrollbars visible on the RHS areas
-Improvements to scrollbar appearance consistency (particularly for Firefox)

And the cons:
-Not as visually polished due to the issues above

esethna commented:
For the record, @kelderek I'm in agreement that the usability improvements outweigh the UI polish issues here. Beyond this PR, I would like to ask help from web platform team take on the effort (or potentially open a HW campaign) around unifying the scroll bar components. cc// @m1lt0n @hmhealey 

My personal experience on Windows is that these scroll bars are acutally preferable to what we have now from a styling perspective. See examples:

Current: 
![image](https://user-images.githubusercontent.com/14333569/198149804-522ec520-0ac9-4192-8598-a2d09b62e19a.png)

This PR: 
![image](https://user-images.githubusercontent.com/14333569/198149867-f9e849b9-8ecd-44e4-b7b5-46e62b067bd1.png)


@matthewbirtch and UX team, do you guys see the UI issues as blockers to merging, or are we okay to merge this, soak on community and then evaluate further if we find new corner cases?

kelderek commented:
@matthewbirtch The thread pane gap should be fixed now (first item in your last post) is fixed now.  As for the issue of the post under the cursor being a different color, for reference the current behavior does the same but it is less noticeable due to the gray scrollbar track:
![image](https://user-images.githubusercontent.com/6126101/198378771-ced53696-a08e-4a9d-9591-b14184a1d951.png)

If leaving the track gray would be better than transparent, that is certainly an option, but then it would be less consistent with the custom scrollbars.

Also added to this commit:
-Changed style naming to be more descriptive and accurate
-Worked out how to properly move the custom whole scrollbar rather than just the thumb portion.  Works out the same from an end-user perspective because the track is transparent, but is a more sensible approach
-Adjusted Firefox styling for for post entry scrollbar that shows up when the post entry box goes over half the screen's height

Here are some other places where the Firefox native scrollbars could be restyled:
-Channel member list on RHS (took a stab at it but haven't figured it out yet because I don't React and TypeScript and scrollbar-width does not inherit)
-Emoji picker, both default and gfycat tabs (haven't tried to adjust)
-@ direct message user picker (haven't tried to adjust)

Webkit browsers aren't a problem with those as they are naturally thinner and inherit the coloring already.

esethna commented:
> If leaving the track gray would be better than transparent

My 2c is to eliminate the track, but I'll leave that call to @matthewbirtch

esethna commented:
@kelderek, Matt and I chatted about this, we're good to move ahead with the rest of the reviews 

Adding @jgilliam17 (QA) and @AshishDhama (front end dev) to help continue the review process

kelderek commented:
Excellent!  Thanks @esethna and @matthewbirtch !  I will be on the lookout in case any additional tweaks need to be done!  Also, the release note I put forth in the beginning is still accurate, but very incomplete now.  Should I revamp that or is it the kind of change that realistically gets rolled up with other small changes into a generic "UX improvements" note?

esethna commented:
@kelderek feel free to edit the release note as needed to keep it accurate :)

kelderek commented:
Updated the release notes and screenshots on the first post

jgilliam17 commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/257645

jgilliam17 commented:
Thanks @kelderek 
Tested, looks good to merge - pending e2es, I'll post report here shortly 
Verified scroll bar border in desktop and browsers.

AshishDhama reviewed (approved):
LGTM, Thanks @kelderek Looking forward to your future contribution. This was indeed a great effort.

jgilliam17 reviewed (approved):
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/257645-dca623d-onprem-ent-webapp-pr-11292-mm-ee-test:dca623d/mochawesome.html) - looks good to merge, no PR related failures.

esethna commented:
@jgilliam17 or others, can you help merge if there's nothing blocking and reviews complete?

esethna commented:
Thanks @kelderek for your great work here!!

kelderek commented:
Thank you @esethna for the encouragement and guidance with this process!

jordanafung commented:
Hi @kelderek 
You were selected as one of our Top Contributors for Hacktoberfest!
Please fill out [this form](https://docs.google.com/forms/d/11pLB3pF1CmOslYqHr7c7bxFWU6QAEAJuiaKgHD1vJtg) to select your swag as thanks!
Thank you for all you do and for making Mattermost even better!

kelderek commented:
Thank you! The form has been filled out.

Jan 5, 2023 13:46:25 Jordana Fung ***@***.***>:

> Hi @kelderek[https://github.com/kelderek]
> You were selected as one of our Top Contributors for Hacktoberfest!
> Please fill out this form[https://docs.google.com/forms/d/11pLB3pF1CmOslYqHr7c7bxFWU6QAEAJuiaKgHD1vJtg] to select your swag as thanks!
> Thank you for all you do and for making Mattermost even better!
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/mattermost/mattermost-webapp/pull/11292#issuecomment-1372599454], or unsubscribe[https://github.com/notifications/unsubscribe-auth/ABOXUFOTLROKTWLIZPZLM43WQ4JFRANCNFSM6AAAAAAQ65LHJQ].
> You are receiving this because you were mentioned.[Tracking image][https://github.com/notifications/beacon/ABOXUFLB2RH7TA3XL4V35ZDWQ4JFRA5CNFSM6AAAAAAQ65LHJSWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSR2A2J4.gif]
>
